### PR TITLE
feat: simbridge support

### PR DESCRIPTION
**Pull request by Benjozork**
## Summary of Changes

* Add simbridge addon definition
* Add dependency support, optional or rmandatory
* Add the ability to declare external apps in publisher defs
* Add  `@/` ref prefix to access current publisher scope
* Add background service support
  * Link to external app def for status query
  * Start/stop support
  * Windows Registry autostart support 
* Start configuration aspect support (unfinished and not working)
* Refactor install logic location and install status slice
* Improve UI for addon state bar (bg service, close external app, install progress)

### CODE REVIEW

File: src/renderer/utils/InstallManager.tsx [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+        fs.removeSync(tempDir);
+    }
+
+    static cancelDownload(download: DownloadItem): void {
+        const abortController = this.abortControllers[download.abortControllerID];
+
+        abortController?.abort();
+    }
+
+    static async uninstallAddon(addon: Addon, publisher: Publisher, showModal: (modal: any) => Promise<boolean>): Promise<void> {
```
Comment (FoxtrotSierra6829) [inline src/renderer/utils/InstallManager.tsx:353]: unexpected any
```suggestion
    static async uninstallAddon(addon: Addon, publisher: Publisher, showModal: (modal: JSX.Element) => Promise<boolean>): Promise<void> {
```

File: src/renderer/utils/ExternalAppsUI.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+export class ExternalAppsUI {
+    /**
+     * Shows a modal presenting the user with running external apps for this addon that must be closed. When all apps are closed,
+     * and the user presses "Confirm", the promise is resolved with `true`. In all other cases, it is resolved with `false`.
+     *
+     * @param addon     the addon
+     * @param publisher the publisher of the addon
+     * @param showModal a function to show a modal and return a promise based on the button clicked
+     */
+    static async ensureNoneRunningForAddon(addon: Addon, publisher: Publisher, showModal: (modal: any) => Promise<boolean>): Promise<boolean> {
```
Comment (FoxtrotSierra6829) [inline src/renderer/utils/ExternalAppsUI.tsx:36]: unexpected any
```suggestion
    static async ensureNoneRunningForAddon(addon: Addon, publisher: Publisher, showModal: (modal: JSX.Element) => Promise<boolean>): Promise<boolean> {
```

File: src/renderer/utils/InstallManager.tsx [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+
+    private static setCurrentSelectedTrack(addon: Addon, track: AddonTrack): void {
+        store.dispatch(setSelectedTrack({ addonKey: addon.key, track: track }));
+    }
+
+    private static setCurrentlyInstalledTrack(addon: Addon, track: AddonTrack): void {
+        store.dispatch(setInstalledTrack({ addonKey: addon.key, installedTrack: track }));
+    }
+
+    static async installAddon(addon: Addon, publisher: Publisher, showModal: (modal: any) => Promise<boolean>): Promise<InstallResult> {
```
Comment (FoxtrotSierra6829) [inline src/renderer/utils/InstallManager.tsx:133]: unexpected any
```suggestion
    static async installAddon(addon: Addon, publisher: Publisher, showModal: (modal: JSX.Element) => Promise<boolean>): Promise<InstallResult> {
```

**Review by FoxtrotSierra6829 — DISMISSED** [review]

**Comment by FoxtrotSierra6829** [comment]
The cancel button on the A32NX page does not reflect the state of the SimBridge dependency and also does not trigger the abort controller for SimBridge. This does only work properly on the SimBridge page. After the installation of the dependencies, the A32NX page does not reflect the state of the current A32NX download. Any other addon than the A32NX cannot be downloaded

**Comment by Benjozork** [comment]
No merge commits on this branch please

**Comment by Benjozork** [comment]
@FoxtrotSierra6829 I have pushed a lot of state management fixes, should resolve the issues with cancelling and installing other addons

File: src/renderer/data.ts [hunk 4]
```diff
@@ -182,6 +276,12 @@ export const defaultConfiguration: Configuration = {
                 //     inop: true,
                 //     inline: true,
                 // },
+                // {
+                //     text: "FBW SimBridge",
+                //     action: 'internal',
+                //     call: 'fbw-local-api-config',
+                //     style: 'fbw-local-api-config',
+                // },
```
Comment (FoxtrotSierra6829) [inline src/renderer/data.ts:284]: think we can delete this, can we?

File: src/renderer/index.scss [hunk 5]
```diff
@@ -105,3 +105,27 @@ a {
 button:focus {
   outline: none;
 }
+
+p {
+    margin-bottom: 0 !important;
```
Comment (FoxtrotSierra6829) [inline src/renderer/index.scss:110]: is the important 100% required?

**Pull request merged**